src/transform/src/join_implementation.rs

```diff
@@ -640,6 +764,25 @@ mod differential {
                 input_mapper,
             )?;
 
+            // Count new arrangements.
+            //
+            // We collect the count for each input, to be used to calculate `new_arrangements` below.
+            let new_input_arrangements: Vec<usize> = orders
+                .iter()
+                .map(|o| {
+                    o.iter()
+                        .skip(1)
+                        .filter_map(|(c, key, input)| {
+                            if c.arranged {
+                                None
+                            } else {
+                                Some((*input, key.clone()))
```

Comment:
Are we using these values that are inside the `Some`?

Reply:
We should be---for the same deduplication of input arrangements happening in delta joins.